aisuite/providers/anthropic_provider.py

```diff
@@ -15,26 +34,195 @@ def __init__(self, **config):
 
         self.client = anthropic.Anthropic(**config)
 
+    def convert_request(self, messages):
+        """Convert framework messages to Anthropic format."""
+        return [self._convert_single_message(msg) for msg in messages]
+
+    def _convert_single_message(self, msg):
+        """Convert a single message to Anthropic format."""
+        if isinstance(msg, dict):
+            return self._convert_dict_message(msg)
+        return self._convert_message_object(msg)
+
+    def _convert_dict_message(self, msg):
+        """Convert a dictionary message to Anthropic format."""
+        if msg["role"] == self.ROLE_TOOL:
+            return self._create_tool_result_message(msg["tool_call_id"], msg["content"])
+        elif msg["role"] == self.ROLE_ASSISTANT and "tool_calls" in msg:
+            return self._create_assistant_tool_message(
+                msg["content"], msg["tool_calls"]
+            )
+        return {"role": msg["role"], "content": msg["content"]}
+
+    def _convert_message_object(self, msg):
+        """Convert a Message object to Anthropic format."""
+        if msg.role == self.ROLE_TOOL:
+            return self._create_tool_result_message(msg.tool_call_id, msg.content)
+        elif msg.role == self.ROLE_ASSISTANT and msg.tool_calls:
+            return self._create_assistant_tool_message(msg.content, msg.tool_calls)
+        return {"role": msg.role, "content": msg.content}
+
+    def _create_tool_result_message(self, tool_call_id, content):
+        """Create a tool result message in Anthropic format."""
+        return {
+            "role": self.ROLE_USER,
+            "content": [
+                {
+                    "type": "tool_result",
+                    "tool_use_id": tool_call_id,
+                    "content": content,
+                }
+            ],
+        }
+
+    def _create_assistant_tool_message(self, content, tool_calls):
+        """Create an assistant message with tool calls in Anthropic format."""
+        message_content = []
+        if content:
+            message_content.append({"type": "text", "text": content})
+
+        for tool_call in tool_calls:
+            tool_input = (
+                tool_call["function"]["arguments"]
+                if isinstance(tool_call, dict)
+                else tool_call.function.arguments
+            )
+            message_content.append(
+                {
+                    "type": "tool_use",
+                    "id": (
+                        tool_call["id"] if isinstance(tool_call, dict) else tool_call.id
+                    ),
+                    "name": (
+                        tool_call["function"]["name"]
+                        if isinstance(tool_call, dict)
+                        else tool_call.function.name
+                    ),
+                    "input": json.loads(tool_input),
+                }
+            )
```

Comment:
Similar thought, converting to either an object or dict at the beginning of this would create a consistency that would not require the `if/else` statements throughout the function call.